src/main/java/de/dennisguse/opentracks/fragments/IntervalsFragment.java

```diff
@@ -211,11 +211,11 @@ public void onTrackUpdated(Track track) {
             getActivity().runOnUiThread(() -> {
                 if (isResumed()) {
                     // Set category.
-                    category = track != null ? track.getCategory() : "";
+                    String category = track != null ? track.getCategory() : "";
 
                     // Set rate label.
-                    boolean reportSpeed = PreferencesUtils.isReportSpeed(sharedPreferences, getContext(), category); //TODO Handle sharedPreferenceChangeListener
-                    viewBinding.intervalRate.setText(reportSpeed ? R.string.stats_speed : R.string.stats_pace);
+                    isReportSpeed = PreferencesUtils.isReportSpeed(sharedPreferences, getContext(), category);
```

Comment:
@dennisguse I think we cannot do it through sharedPreferencesChangeListener because we are looking for category change. And it's not in preferences but in activity edit, right?

Reply:
if it works, go for it ;)
we may or may not improve this later